src/utils/vc.js

```diff
@@ -114,17 +166,25 @@ export async function checkRevocationStatus(credential, revocationApi) {
   if (!revocationApi.dock) {
     throw new Error('Only Dock revocation support is present as of now.');
   } else {
-    if (!hasDockRevocation(credential)) {
-      return { verified: false, error: 'The credential status does not have the format required by Dock' };
-    }
+    const statuses = await getCredentialStatuses(credential);
     const dockAPI = revocationApi.dock;
-    const regId = credential.credentialStatus.id.slice(DockRevRegQualifier.length);
-    // Hash credential id to get revocation id
     const revId = getDockRevIdFromCredential(credential);
-    const revocationStatus = await dockAPI.revocation.getIsRevoked(regId, revId);
-    if (revocationStatus) {
-      return { verified: false, error: 'Revocation check failed' };
+    for (let i = 0; i < statuses.length; i++) {
+      const status = statuses[i];
+
+      if (!hasDockRevocation(status)) {
+        return { verified: false, error: 'The credential status does not have the format required by Dock' };
+      }
+
+      const regId = status['@id'].slice(DockRevRegQualifier.length);
+
+      // Hash credential id to get revocation id
+      const revocationStatus = await dockAPI.revocation.getIsRevoked(regId, revId); // eslint-disable-line
+      if (revocationStatus) {
+        return { verified: false, error: 'Revocation check failed' };
+      }
     }
+
     return { verified: true };
```

Comment:
but if the credential has multiple credential statuses that must be checked in future, this is a bad idea. we only support dock revocation for now but do we not intend to support more? what if the dock revocation status is 1st but theres other 2nd, 3rd statuses to look at? we should inform the user that the statuses provided are invalid compared to what we expect, even if there is one valid one

Reply:
We expect each revocation to be Dock with this `hasDockRevocation` so if there is any other kind, it would fail anyway.

Reply:
yes but if you moved verified: true into inside the loop, it won't fail if the credential has another status object after the dock one. i think it should, as it would if it had only that status and not a dock one

Reply:
imo this would result in a false positive

Reply:
hasDockRevocation now checks individual statuses, rather than all of them before the loop. if this wasnt the case then id agree with you

Reply:
@lovesh, I don't think there is anything stopping someone from crafting a credential with multiple Dock "credentialStatus"es.